### PR TITLE
DSA searching&LinkedList

### PR DESCRIPTION
**Pull request by Abishek3896**
For your review

### CODE REVIEW

File: Week-14/Binary_Search_1/Get_Square_root.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+function getSquareRoot(number) {
+  let start = 0;
+  let end = number;
+  let beta = 0.000001;
```
Comment (svashish305) [inline Week-14/Binary_Search_1/Get_Square_root.js:4]: refactor beta to a better meaningful name, like precisionNumber or so

File: Week-14/Binary_Search_1/Get_last_occurence.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      isAnswer = true;
+      start = median + 1;
+    } else {
+      start = median + 1;
+    }
+  }
+  return isAnswer ? answer : -1;
+}
+
+//console.log(getLastOccurence([1, 2, 3, 3, 5], 3));
```
Comment (svashish305) [inline Week-14/Binary_Search_1/Get_last_occurence.js:21]: Possible refactor (optional)

`
function lastOc(a, k) {
  let n = a.length;
  let l = 0, h = n - 1, m;
  while (l <= h) {
    m = Math.floor((l + h) / 2);
    if (a[m] === k) {
      if (m === n - 1) return m;
      if (a[m + 1] === k) l = m + 1;
      else return m;
    } else if (a[m] > k) {
      h = m - 1;
    } else {
      l = m + 1;
    }
  }
  return -1;
}
`

File: Week-14/Binary_Search_1/K-th_Element.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+function kthElement(arr1, arr2, k) {
+  let i = 0;
+  let j = 0;
+  let mergedArr = [];
+  while (i < arr1.length && j < arr2.length) {
+    if (arr1[i] <= arr2[j]) {
+      mergedArr.push(arr1[i]);
+      mergedArr.push(arr2[j]);
+      i += 1;
```
Comment (svashish305) [inline Week-14/Binary_Search_1/K-th_Element.js:9]: in js, j++ works as well

File: Week-14/Binary_Search_2/Search_a_2D_Matrix [hunk 4]
```diff
@@ -0,0 +1,46 @@
+function findTarget(row, column, arr, target) {
```
Comment (svashish305) [inline Week-14/Binary_Search_2/Search_a_2D_Matrix]: rename this file to have a .js extension

File: Week-14/Link_List_1/Is_LL_Palindrome.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+var isPalindrome = function (head) {
```
Comment (svashish305) [inline Week-14/Link_List_1/Is_LL_Palindrome.js:7]: [mandatory]: declare the linkedlist class

File: Week-14/Link_List_1/Remove_Duplicates.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+function removeDuplicates(head) {
```
Comment (svashish305) [inline Week-14/Link_List_1/Remove_Duplicates.js:7]: [mandatory]: declare the linkedlist class

File: Week-14/Link_List_1/Reverse_every_k.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+function reverseEveryK(head, k) {
```
Comment (svashish305) [inline Week-14/Link_List_1/Reverse_every_k.js:7]: [mandatory]: declare the linkedlist class

File: Week-14/Link_List_1/merge_two_sorted.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
+function mergeTwolist(list1, list2) {
```
Comment (svashish305) [inline Week-14/Link_List_1/merge_two_sorted.js:7]: [mandatory]: declare the linkedlist class

**Review by svashish305 — CHANGES_REQUESTED** [review]
handle the mandatory changes

File: Week-14/Link_List_1/merge_two_sorted.js (hunk 8 above)
Comment (Abishek3896) [reply Week-14/Link_List_1/merge_two_sorted.js:7]: I have declared the class

File: Week-14/Binary_Search_2/Search_a_2D_Matrix (hunk 4 above)
Comment (Abishek3896) [reply Week-14/Binary_Search_2/Search_a_2D_Matrix]: Done

File: Week-14/Link_List_1/Is_LL_Palindrome.js (hunk 5 above)
Comment (Abishek3896) [reply Week-14/Link_List_1/Is_LL_Palindrome.js:7]: declared

File: Week-14/Link_List_1/Remove_Duplicates.js (hunk 6 above)
Comment (Abishek3896) [reply Week-14/Link_List_1/Remove_Duplicates.js:7]: declared

File: Week-14/Link_List_1/Reverse_every_k.js (hunk 7 above)
Comment (Abishek3896) [reply Week-14/Link_List_1/Reverse_every_k.js:7]: declared

**Review by svashish305 — APPROVED** [review]

**Pull request closed without merging**